tests/ImageSharp.Tests/Processing/Overlays/VignetteTest.cs

```diff
@@ -0,0 +1,70 @@
+﻿// <copyright file="VignetteTest.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.Tests.Processing.Overlays
+{
+    using ImageSharp.PixelFormats;
+
+    using Xunit;
+
+    public class VignetteTest : FileTestBase
+    {
+        [Theory]
+        [WithFileCollection(nameof(DefaultFiles), StandardPixelType)]
+        public void ImageShouldApplyVignetteFilter<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            using (Image<TPixel> image = provider.GetImage())
+            {
+                image.Vignette()
+                    .DebugSave(provider, null, Extensions.Bmp);
+            }
+        }
+
+        [Theory]
+        [WithFileCollection(nameof(DefaultFiles), StandardPixelType)]
+        public void ImageShouldApplyVignetteFilterColor<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            using (Image<TPixel> image = provider.GetImage())
+            {
+                image.Vignette(NamedColors<TPixel>.Orange)
+                    .DebugSave(provider, null, Extensions.Bmp);
+            }
+        }
+
+        [Theory]
+        [WithFileCollection(nameof(DefaultFiles), StandardPixelType)]
+        public void ImageShouldApplyVignetteFilterRadius<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            using (Image<TPixel> image = provider.GetImage())
+            {
+                image.Vignette(image.Width / 4F, image.Height / 4F)
+                    .DebugSave(provider, null, Extensions.Bmp);
+            }
+        }
+
+        [Theory]
+        [WithFileCollection(nameof(DefaultFiles), StandardPixelType)]
+        public void ImageShouldApplyVignetteFilterInBox<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            using (Image<TPixel> source = provider.GetImage())
+            using (var image = new Image<TPixel>(source))
+            {
+                var bounds = new Rectangle(10, 10, image.Width / 2, image.Height / 2);
+
+                image.Vignette(bounds)
+                    .DebugSave(provider, null, Extensions.Bmp);
+
+                // Draw identical shapes over the bounded and compare to ensure changes are constrained.
+                image.Fill(NamedColors<TPixel>.HotPink, bounds);
```

Comment:
Wouldn't it be better to DRY all these Fill+Fill+CheckSimilarity calls away into a common `EnsureProcessorChangesAreConstrained()` method somewehere?